### ISSUE TITLE
Global variable

### ISSUE THREAD

**Issue by navjotk** [opesci/inversion#24]
The variable ``xrec`` [here](https://github.com/opesci/inversion/blob/master/python/AcousticWave2D.py#L388) seems to be related to the position of the receiver. If that is correct, would it be better to have it as a field inside the data object, instead of being a global?

**Comment by mloubout** [comment]
Yes it would, this was hard coded here to test the propagator, but in practice, all data related fields should come from the data container.
